### PR TITLE
implement noiseSFTs option for Writer

### PR DESCRIPTION
**Pull request by Rodrigo-Tenorio**
Ref. #40 

- [x] Implement noise SFTs for Writer and sons
- [x] Implement test

P. S.
@dbkeitel I'm not sure if this is the right way to do this: This builds upon #39. After getting #39 merged into master, I'll properly rebase this one on top of master from the point of divergence.

### CODE REVIEW

**Comment by dbkeitel** [comment]
Will need a manual rebase now.

File: pyfstat/make_sfts.py [hunk 1]
```diff
@@ -370,9 +373,33 @@ def run_makefakedata(self):
...
         cl_mfd.append('--outLabel="{}"'.format(self.label))
-        cl_mfd.append("--IFOs={}".format(self.IFOs))
+
         if self.add_noise:
+
+            if self.noiseSFTs is not None and self.SFTWindowType is None:
+                raise ValueError(
+                    "SFTWindowType is required when using noiseSFTs. "
+                    "Please, make sure you understand the window function used "
+                    "to produced noiseSFTs."
```
Comment (dbkeitel) [inline pyfstat/make_sfts.py:383]: minor typo: produced->produce

File: pyfstat/make_sfts.py [hunk 2]
```diff
@@ -370,9 +373,33 @@ def run_makefakedata(self):
         cl_mfd.append("--outSingleSFT=TRUE")
         cl_mfd.append('--outSFTdir="{}"'.format(self.outdir))
         cl_mfd.append('--outLabel="{}"'.format(self.label))
-        cl_mfd.append("--IFOs={}".format(self.IFOs))
+
         if self.add_noise:
+
```
Comment (dbkeitel) [inline pyfstat/make_sfts.py:378]: could you remove this blank line and 389 too so the block hierarchy stays cleaner

File: pyfstat/make_sfts.py [hunk 3]
```diff
@@ -370,9 +373,33 @@ def run_makefakedata(self):
...
+            elif self.noiseSFTs is not None:
+                cl_mfd.append('--noiseSFTs="{}"'.format(self.noiseSFTs))
+            else:
+                cl_mfd.append("--IFOs={}".format(self.IFOs))
+
+            logging.warning(
+                "Gaussian noise *is added* by default due to "
+                "consistency issues. Please, make sure sqrtSX value is "
+                "consistent with what you intend to do (0=No Gaussian noise). "
+                "Currently, sqrtSX={}".format(self.sqrtSX)
```
Comment (dbkeitel) [inline pyfstat/make_sfts.py:394]: That whole situation may require some further thought, I'm not so sure if the separate `add_noise` flag really makes sense either. But for now it's a good idea to put in that warning and if I end up unhappy with the sqrtSX situation, I'll make a separate issue/PR. Thanks.

**Comment by dbkeitel** [comment]
For the record, the lalsuite fix is here: https://git.ligo.org/lscsoft/lalsuite/-/merge_requests/1354
But we decided to implement our own safety check on top of it, to be also safe with old versions. (New LALSuite release should be imminent, but can still be delayed.)

**Comment by Rodrigo-Tenorio** [comment]
For the sake of completeness I'll give some clarification on 20acefb:

- Makefakedata_v5's kerfuffle
  - `SFTWindowType` was supposed to be required (and it actually was at _v4) but somehow such requirement flow over everyone's head and wasn't being enforced *at all*.
  - PyFstat has been updated to enforce this requirement. It is useful to make the user aware of the choices being done.

- The issue with sqrtSX
  - sqrtSX is being (wrongly) used as two different things at different points of this code: As a way of specifying Gaussian noise at MFD and as a PSD measure at ComputeFstat. 
  - This is problematic, since *no Gaussian noise* is represented as sqrtSX=0 in MFD, but as sqrtSX=1 in ComputeFstat.
  - Short term solution: Warn the user to set sqrtSX=0 when creating SFTs.

- Test
  - Improved naming conventions prevents name clashing between different stages.

File: pyfstat/make_sfts.py (hunk 3 above)
Comment (Rodrigo-Tenorio) [reply pyfstat/make_sfts.py:394]: I'm not sure how many things will break down when removing `add_noise`. Agree with doing a separate issue/PR.

Comment (dbkeitel) [reply pyfstat/make_sfts.py:394]: see #50

**Pull request merged**